src/main/scala/CDS/FileCollection.scala

```diff
@@ -0,0 +1,159 @@
+package CDS
+import scala.io.Source
+import java.net.URI
+
+case class FileCollection(mediaFile:String,inmetaFile:String,metaFile:String,xmlFile:String,dsLocation:URI,tempFile:String){
+  def replace(xmediaFile:Option[String],xinmetaFile:Option[String],xmetaFile:Option[String],xxmlFile:Option[String]):FileCollection = {
```

Comment:
case classes already have a `copy()` method, which unless I am mistaken essentially implements this functionality, e.g. line 103:

```scala
            Some(fileCollection.replace(Some(media),Some(inmeta),Some(meta),Some(xml)))
```
could just be:
```scala
        fileCollection.copy(media, inmeta, xml) // other args are optional, use named args to skip
```